bitnami/tomcat/templates/_helpers.tpl

```diff
@@ -34,3 +43,38 @@ Expand the name of the chart.
 {{- define "tomcat.pvc" -}}
 {{- coalesce .Values.persistence.existingClaim (include "common.names.fullname" .) -}}
 {{- end -}}
+
+{{/*
+Return the proper CATALINA_OPTS value
+*/}}
+{{- define "tomcat.catalinaOpts" -}}
+  {{- printf "%s %s" .Values.catalinaOpts .Values.metrics.jmx.catalinaOpts  | trim  -}}
+{{- end -}}
+
+{{/*
+Return the proper JMX exporter image name
+*/}}
+{{- define "tomcat.metrics.jmx.image" -}}
+{{ include "common.images.image" (dict "imageRoot" .Values.metrics.jmx.image "global" .Values.global) }}
+{{- end -}}
+
+{{/*
+Return the tomcat jmx configuration configmap
+*/}}
+{{- define "tomcat.metrics.jmx.configmapName" -}}
+{{- if .Values.metrics.jmx.existingConfigmap -}}
+    {{- printf "%s" (tpl .Values.metrics.jmx.existingConfigmap $) -}}
+{{- else -}}
+    {{- printf "%s-jmx-configuration" (include "tomcat.fullname" .) -}}
+{{- end -}}
+{{- end -}}
+{{/*
+Check if jmx metrics is enabled, then either metrics.jmx.config or metrics.jmx.existingConfigmap must be set.
+*/}}
+{{- define "tomcat.checkJmxConfig" -}}
+  {{- if .Values.metrics.jmx.enabled -}}
+    {{- if and (not .Values.metrics.jmx.config)  (not .Values.metrics.jmx.existingConfigmap) -}}
+      {{- printf "%s" "\nJMX CONFIG ERROR: You must provide JMX configuration by set either metrics.jmx.config or metrics.jmx.existingConfigmap" | fail -}}
+    {{- end -}}
+  {{- end -}}
+{{- end -}}
```

Comment:
We usually do this kind of operations following a standard. Could you please follow the same approach as here https://github.com/bitnami/charts/blob/d58fc96d78129724a0f72beb297f76926f04ac6e/bitnami/cassandra/templates/_helpers.tpl#L74